### PR TITLE
Feature/cd flow

### PR DESCRIPTION
**Pull request by KerstenBreuer**
- adds a cd pipeline that upon every push to the dev (and eventually int) branch:
  - generates a version of the following pattern: 
    - `<last tag of branch>-<commit count since tag>-<short commit hash>-<branch>`
    - if no tag was set yet it will use 0.0.0
    - example if the last tag was `0.5.2`, there were 17 commits since that tag, and the last commit has the short hash `776c95c` and the branch that triggered the CD was `dev`: `0.5.2-17-776c95c-dev`
    - please note, the tags/releases but not these commit-based versions are following semantic versioning
  - a docker container is built and published to docker hub
    - the docker repo name is the same as the github repo name
    - the commit version generated before serves as image tag
  - update the deployment repo:
    - send a webhook to the github repository containing deployment helm charts
    - parameters are:
      - "chart_name": the name of this github repo
      - "new_version": commit version as generated above
    - the corresponding helm chart will be updated with the new commit version as the `appVersion`, the `appVersion` is used as docker image tag 
- add a Dockerfile for deployment
- pin versions in the setup.py

I am looking forward to your suggestions :)

(The merge of this PR will trigger the CD workflow for the first time, so this will serve as initial test.)

### CODE REVIEW

**Review by MoritzHahn1337 — APPROVED** [review]

**Review by deepakunni3 — APPROVED** [review]

**Comment by deepakunni3** [comment]
Thanks @KerstenBreuer for the well thought out flow for continuous delivery 👍

**Comment by KerstenBreuer** [comment]
Thanks @deepakunni3 :)

File: .devcontainer/dev_install [hunk 1]
```diff
@@ -3,6 +3,10 @@
...
-# install client in edit mode 
-# with testing dependencies:
-pip install -e ".[testing]"
+# install in edit mode
+# use future way of installing packages
+# install with testing dependencies
+pip install \
+    --use-feature=in-tree-build \
+    -e \
+    ".[testing]"
```
Comment (ac-jorellanaf) [inline .devcontainer/dev_install:12]: Add newline

File: .github/workflows/dev_cd.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            "https://api.github.com/repos/${DEPLOYMENT_CONFIG_REPO}/dispatches" \
+            -H 'Accept: application/vnd.github.everest-preview+json' \
+            -u '${{ secrets.DEPLOYMENT_UPDATE_TOKEN }}' \
+            --data '{
+              "event_type": "new_app_version", 
+              "client_payload": {
+                "chart_name": "${{ github.event.repository.name }}", 
+                "new_app_version": "${{ needs.get_commit_version.outputs.version }}"
+              } 
+            }'
```
Comment (ac-jorellanaf) [inline .github/workflows/dev_cd.yml:83]: Add newline.

**Review by ac-jorellanaf — APPROVED** [review]
I hope you tested out the CD behaviour. Something always goes badly when trying to automate!
Everything seems in good order, though I am not familiar with CD, or the specific sytax.

**Comment by KerstenBreuer** [comment]
Thanks, @ac-jorellanaf, I tested it on another repo, there it worked fine. However, there is unfortunately only one way how to test it on this repo and that is to merge it. So let's keep our fingers crossed. ;)

**Comment by KerstenBreuer** [comment]
@alamgirmani I hope that it's fine that I merge it now, so that we get this tested.

**Pull request merged**